ssh/ssh.go

```diff
@@ -20,22 +21,71 @@ type Key struct {
 	Fingerprint string // Name to associate with fingerprint
 }
 
-// GenerateKey will create a private key and fingerprint
-func GenerateKey(secrets *v1.Secret, key Key) {
+// RecordKeyInfo records priave key or fingerprint names for later generation
+func RecordKeyInfo(keys map[string]Key, configVar *model.ConfigurationVariable) error {
+	if len(configVar.Generator.ID) == 0 {
+		return fmt.Errorf("Config variable `%s` has no ID value", configVar.Name)
+	}
+	if configVar.Generator.Type != model.GeneratorTypeSSH {
+		return fmt.Errorf("Config variable `%s` does not have a valid SSH generator type", configVar.Name)
+	}
+
+	// Get or create the key from the map, there should always be
+	// a pair of private keys and fingerprints
+	key := keys[configVar.Generator.ID]
+
+	switch configVar.Generator.ValueType {
+	case model.ValueTypeFingerprint:
+		if len(key.Fingerprint) > 0 {
+			return fmt.Errorf("Multiple variables define fingerprints name for SSH id `%s`", configVar.Generator.ID)
```

Comment:
It might be a good idea to print out the two config names (`key.Fingerprint` and `configVar.Name`) so it's easier to track down

Reply:
idk, this should never happen because the role manifest should have been verified by fissile. And if this happens, it is easy enough to grep the role manifest for all variables with this id:

```
grep -C 5 "id: router_cert" container-host-files/etc/scf/config/role-manifest.yml
```
Do you really think this is worth it?

Reply:
Meh, it's probably fine as-is